packages/lib/CalendarService.ts

```diff
@@ -300,133 +303,127 @@ export default abstract class BaseCalendarService implements Calendar {
-      await Promise.all(
-        selectedCalendars
-          .filter((sc) => ["caldav_calendar", "apple_calendar"].includes(sc.integration ?? ""))
-          .map((sc) =>
-            fetchCalendarObjects({
-              urlFilter: (url: string) => this.isValidFormat(url),
-              calendar: {
-                url: sc.externalId,
-              },
-              headers: this.headers,
-              expand: true,
-              timeRange: {
-                start: startISOString,
-                end: new Date(dateTo).toISOString(),
-              },
-            })
-          )
-      )
-    ).flat();
 
+    const objects = await this.fetchObjectsWithOptionalExpand({
+      selectedCalendars,
+      startISOString,
+      dateTo,
+      headers: this.headers,
+    });
     const events: { start: string; end: string }[] = [];
-
     objects.forEach((object) => {
       if (object.data == null || JSON.stringify(object.data) == "{}") return;
-
-      const jcalData = ICAL.parse(sanitizeCalendarObject(object));
-      const vcalendar = new ICAL.Component(jcalData);
-      const vevent = vcalendar.getFirstSubcomponent("vevent");
-
-      // if event status is free or transparent, return
-      if (vevent?.getFirstPropertyValue("transp") === "TRANSPARENT") return;
-
-      const event = new ICAL.Event(vevent);
-      const dtstart: { [key: string]: string } | undefined = vevent?.getFirstPropertyValue("dtstart");
-      const timezone = dtstart ? dtstart["timezone"] : undefined;
-      // We check if the dtstart timezone is in UTC which is actually represented by Z instead, but not recognized as that in ICAL.js as UTC
-      const isUTC = timezone === "Z";
-      const tzid: string | undefined = vevent?.getFirstPropertyValue("tzid") || isUTC ? "UTC" : timezone;
-      // In case of icalendar, when only tzid is available without vtimezone, we need to add vtimezone explicitly to take care of timezone diff
-      if (!vcalendar.getFirstSubcomponent("vtimezone") && tzid) {
-        const timezoneComp = new ICAL.Component("vtimezone");
-        timezoneComp.addPropertyWithValue("tzid", tzid);
-        const standard = new ICAL.Component("standard");
-        // get timezone offset
-        const tzoffsetfrom = buildUtcOffset(dayjs(event.startDate.toJSDate()).tz(tzid, true).utcOffset());
-        const tzoffsetto = buildUtcOffset(dayjs(event.endDate.toJSDate()).tz(tzid, true).utcOffset());
-        // set timezone offset
-        standard.addPropertyWithValue("tzoffsetfrom", tzoffsetfrom);
-        standard.addPropertyWithValue("tzoffsetto", tzoffsetto);
-        // provide a standard dtstart
-        standard.addPropertyWithValue("dtstart", "1601-01-01T00:00:00");
-        timezoneComp.addSubcomponent(standard);
-        vcalendar.addSubcomponent(timezoneComp);
+      let vcalendar: ICAL.Component;
+      try {
+        const jcalData = ICAL.parse(sanitizeCalendarObject(object));
+        vcalendar = new ICAL.Component(jcalData);
```

Comment:
We place ICAL.parse in a `try-catch` as .parse function throws error when the object is not of iCalendar 2.0 standard, and it ends up breaking the working calendars as well. The `try-catch` fails it without breaking the flow.